AsyncUsageAnalyzers/AsyncUsageAnalyzers/Helpers/MethodSymbolExtensions.cs

```diff
@@ -25,7 +25,8 @@ public static bool HasAsyncSignature(this IMethodSymbol symbol, bool treatAsyncV
             if (!symbol.IsAsync)
             {
                 // This check conveniently covers Task and Task<T> by ignoring the `1 in Task<T>.
-                if (!string.Equals(nameof(Task), symbol.ReturnType?.Name, StringComparison.Ordinal))
+                if (!string.Equals(nameof(Task), symbol.ReturnType?.Name, StringComparison.Ordinal)
+                    && !string.Equals("ValueTask", symbol.ReturnType?.Name, StringComparison.Ordinal))
```

Comment:
Why not allow any type whose GetAwaitInfo suggests the type is awaitable?

Reply:
❓ Is there an easy way to do this using existing Roslyn APIs? Otherwise I was planning to leave it as a follow-up change.

Reply:
I'm pretty sure it's been there since the earliest versions. Let me find it...

Reply:
Ok, never mind. The API I was thinking of is `SemanticModel.GetAwaitExpressionInfo` which accepts an await expression rather than a type that may be awaited on.

Reply:
I'll file a follow-up issue to go based on `AsyncMethodBuilderAttribute`:

https://github.com/dotnet/coreclr/blob/ac8f8e52c49b0ffa712d68730a41883d281bea01/src/System.Private.CoreLib/shared/System/Threading/Tasks/ValueTask.cs#L58

Reply:
Rather than base it on whether a method's return type is compatible with the `async` method declaration modifier, why not just expect/require an Async suffix anytime the returned type is awaitable? For example, people may return custom awaitables from non-async methods that nevertheless dictate that the calling pattern for the method is to await the result. `SwitchToMainThreadAsync` is an example of this. We have several other custom awaitables returned from various Async-suffixed methods as well.

Reply:
I filed #73 and #74 to cover both cases.

Reply:
@sharwell there is http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Shared/Extensions/ISymbolExtensions.cs,882

Or you can roll your own:
```cs
internal static bool IsAwaitable(this ITypeSymbol type)
{
    return type.TryFindFirstMethod("GetAwaiter", x => x.Parameters.Length == 0, out var method) &&
           method.ReturnType is ITypeSymbol returnType &&
           returnType.TryFindFirstMethod("GetResult", x => x.Parameters.Length == 0, out _);
}
```

The above does not check for extension methods and there can be more checks on the awaiter type.